pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1696,6 +1705,20 @@ private IdealState updatePauseStatusInIdealState(String tableNameWithType, boole
     return updatedIdealState;
   }
 
+  public boolean updateStorageQuotaExceededInIdealState(String tableNameWithType, boolean quotaExceeded) {
+    IdealState is = getIdealState(tableNameWithType);
+    if (is.getRecord().getBooleanField(IS_QUOTA_EXCEEDED, false) != quotaExceeded) {
+      HelixHelper.updateIdealState(_helixManager, tableNameWithType, idealState -> {
+        ZNRecord znRecord = idealState.getRecord();
+        znRecord.setSimpleField(IS_QUOTA_EXCEEDED, Boolean.valueOf(quotaExceeded).toString());
+        return new IdealState(znRecord);
+      }, RetryPolicies.noDelayRetryPolicy(1));
```

Comment:
Why not allow retry here?

Reply:
No strong reason to not retry. Was keeping it inline with the table pause IS update. 
Right now in case of failure the next segment commit or validation job, whichever runs first, will update the IS.
Let me know if you have any magic number for retry in mind, maybe 3?